### ISSUE TITLE
Usage Error for (( x = 1)) vs ( x == 1))

### ISSUE THREAD

**Issue by codersteinersky** [koalaman/shellcheck#2518]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# Incorrect code
row=2; (( row = 1 )); echo $?; echo $row
# 0
# 1

# Corrected code
row=2; (( row == 1 )); echo $?; echo $row
1
2

```

#### Here's what shellcheck currently says:
```
No issues detected!
```
#### Here's what I wanted or expected to see:

With spaces I expected the single "=" to act as an equality comparison like it does with "[" or "[[" but with "((" this apparently works as an assignment.  It'd be nice if this were flagged as a probable usage error.

This is my first enhancement request.  Think it's legit but either way shellcheck is a FANTASTIC resource.  Thanks for creating it!!

**Comment by codersteinersky** [comment]
A bit more context.  In my actual script the statement was a conditional "if (( row = 1 )); then"
